DMD.h

```diff
@@ -0,0 +1,177 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2021, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+// Description:Computes the DMD algorithm on the given snapshot matrix.
+
+#ifndef included_DMD_h
+#define included_DMD_h
+
+#include <vector>
+#include <complex>
+
+namespace CAROM {
+
+class Matrix;
+class Vector;
+
+class DMD
+{
+public:
+
+    /**
+     * @brief Constructor.
+     */
+    DMD(int dim);
+
+    /**
+     * @brief Sample the new state, u_in.
+     *
+     * @pre u_in != 0
+     *
+     * @param[in] u_in The state at the specified time.
+     *
+     * @return True if the sampling was successful.
```

Comment:
What does `false` return value of `takeSample` mean, and how should it be handled? This should be documented.